src/algebra/group/pi.lean

```diff
@@ -137,6 +137,59 @@ variables {α β γ : Type*}
 
 end instance_lemmas
 
+section
+
+variables [decidable_eq I]
+
+/-- Extend a function from `{i}ᶜ : set I` to `I`. -/
+def extend_compl_single (i : I) (y : f i) (g : Π j : ({i}ᶜ : set I), f j) :
```

Comment:
Is there an advantage to this formulation vs the one that avoids subtypes:
```suggestion
def extend_compl_single (i : I) (y : f i) (g : Π j, j ≠ i → f j) :
```

Reply:
In the `stokes-new` branch I use it to embed a face of a rectangular box into the box. So, I need functions of the form `ι → R`. I'm not sure whether I should use `extend_compl_single`, or switch to `fin n` and use `fin.insert_nth`.